larray/inout/hdf.py

```diff
@@ -1,32 +1,603 @@
+        #          To minimize explicit conversion troubles when writing, especially when creating data sets
+        #          from existing Python objects, string objects are implicitly cast to non-Unicode byte strings
+        #          for HDF5 storage by default.
+        #          To avoid such problem, one way is to use the VLArray class and dump unicode string arrays
+        #          as object arrays.
+        if dtype.kind == 'O':
+            vlarr = self._handle.create_vlarray(group, name=name, filters=self._filters, atom=tables.ObjectAtom())
+            vlarr.append(data)
+        else:
+            if dtype.kind == 'U':
+                data = np.char.encode(data, 'utf-8')
+            self._handle.create_carray(group, name=name, obj=data, filters=self._filters)
+
+    def _write_obj(self, group, value, **kwargs):
+        if isinstance(value, LArray):
+            attrs = group._v_attrs
+            attrs['type'] = 'Array'
+            # dump axes
+            for axis in value.axes:
+                axis_key = 'axis_{}'.format(value.axes.axis_id(axis))
+                axis_group = self._handle.create_group(group, axis_key)
+                self._write_obj(axis_group, axis)
+            # dump data
+            self._dump_data(group, name='data', data=value.data, attrs=attrs)
+            # dump metadata
+            self._write_obj(group, value.meta)
+        elif isinstance(value, Axis):
+            attrs = group._v_attrs
+            attrs['type'] = 'Axis'
+            attrs['name'] = value.name
+            attrs['wildcard'] = value.iswildcard
+            self._dump_data(group, name='labels', data=value.labels, attrs=attrs)
+        elif isinstance(value, Group):
+            axis_key = kwargs.pop('axis_key', None)
+            if axis_key is None:
+                if value.axis.name is None:
+                    raise ValueError(
+                        "Argument axis_key must be provided explicitly if the associated axis is anonymous")
+                axis_key = value.axis.name
+            if self.get_node(axis_key) is None:
+                self._put(axis_key, value.axis)
+            attrs = group._v_attrs
+            attrs['type'] = 'Group'
+            attrs['name'] = value.name
+            attrs['axis_key'] = axis_key
+            self._dump_data(group, name='key', data=value.eval(), attrs=attrs)
+        elif isinstance(value, Metadata):
+            if len(value):
+                group._v_attrs['metadata'] = value
+        else:
+            warnings.warn('{}: Type {} is currently not supported'.format(group._v_name, type(value)))
+
+    def _write_group(self, key, value, **kwargs):
+        # remove the group if exists already
+        group = self.get_node(key)
+        if group is not None:
+            self._handle.remove_node(group, recursive=True)
+        paths = key.split('/')
+        # recursively create the parent groups
```

Comment:
doesn't the createparents argument do this?
https://www.pytables.org/usersguide/libref/file_class.html#tables.File.create_group